infrastructure/tests/test_data_import.py

```diff
@@ -124,3 +187,111 @@ def test_year_error(self):
+        self.assertEquals(project.longitude, 0.0)
+        self.assertEquals(project.latitude, 0.0)
+
+        self.assertEquals(Expenditure.objects.all().count(), 5)
+        expenditure = Expenditure.objects.get(amount=340609.00)
+        self.assertEquals(expenditure.project.project_description, "P-CNIN FURN & OFF EQUIP")
+        self.assertEquals(expenditure.budget_phase.name, "Audited Outcome")
+        self.assertEquals(expenditure.financial_year.budget_year, "2017/2018")
+
+
+    def test_update_project(self):
+        """Scope of Test: With an existing project import a new project with the same composite key and check that non-key fields are updated"""
+
+        geography = Geography.objects.get(geo_code="BUF")
+        utils.load_file(geography, generate_mock_data(), "2019/2020")
+
+        project = Project.objects.get(project_description="P-CNIN FURN & OFF EQUIP")
+        self.assertEquals(project.project_description, "P-CNIN FURN & OFF EQUIP")
+        self.assertEquals(project.project_number, "PC002003005_00002")
+        self.assertEquals(project.project_type, "New")
+        self.assertEquals(project.mtsf_service_outcome, "An efficient, effective and development-oriented public service")
+        self.assertEquals(project.iudf, "Growth")
+        self.assertEquals(project.own_strategic_objectives, "OWN MUNICIPAL STRATEGIC OBJECTIVE")
+        self.assertEquals(project.asset_class, "Furniture and Office Equipment")
+        self.assertEquals(project.asset_subclass, "")
+        self.assertEquals(project.ward_location, "Administrative or Head Office")
+        self.assertEquals(project.longitude, 0.0)
+        self.assertEquals(project.latitude, 0.0)
+
+        utils.load_file(geography, mock_data_existing_project(), "2020/2021")
+        self.assertEquals(BudgetPhase.objects.all().count(), 5)
+        self.assertEquals(Project.objects.all().count(), 1)
+
+        project = Project.objects.get(project_description="P-CNIN FURN & OFF EQUIP")
+        self.assertEquals(project.project_type, "Renewal")
+        self.assertEquals(project.mtsf_service_outcome, "A project update")
+        self.assertEquals(project.iudf, "Governance")
+        self.assertEquals(project.own_strategic_objectives, "TO BE CORRECTED")
+        self.assertEquals(project.asset_class, "Furniture and Office")
+        self.assertEquals(project.asset_subclass, "Equipment")
+        self.assertEquals(project.ward_location, "Head Office")
+        self.assertEquals(project.longitude, 1.0)
+        self.assertEquals(project.latitude, 2.0)
+
+        self.assertEquals(Expenditure.objects.all().count(), 8)
+        expenditure = Expenditure.objects.get(amount=440609.00)
+        self.assertEquals(expenditure.project.project_description, "P-CNIN FURN & OFF EQUIP")
+        self.assertEquals(expenditure.budget_phase.name, "Audited Outcome")
+        self.assertEquals(expenditure.financial_year.budget_year, "2018/2019")
```

Comment:
same as above
- we wanna be sure that the expenditure items are associated wit the project, and we can do that via foreign key
- we wanna be sure that the ones that exist (that were checked above) had their amount updated, and the new ones were created (like audit outcome here)

Reply:
test_new_project() now uses foreign keys to confirm new expenditures are associated with the new project. For other tests, there is only one project so all the project_id foreign keys for expenditures are the same